src/Core/Helpers.cs

```diff
@@ -60,21 +60,37 @@ private static HttpRequestMessage CreateProxyHttpRequest(this HttpContext contex
 
             var requestMessage = new HttpRequestMessage();
             var requestMethod = request.Method;
+
+            // Copy the request headers.
+            if (requestMessage.Content != null)
+            {
+                foreach (var header in request.Headers)
+                {
+                    if (!requestMessage.Headers.TryAddWithoutValidation(header.Key, header.Value.ToArray()))
+                    {
+                        requestMessage.Content?.Headers.TryAddWithoutValidation(header.Key, header.Value.ToArray());
+                    }
+                }
+            }

```

Comment:
Why did you move this?  Also, please remove braces: I know some people like that standard, but I do not.

Reply:
For some reason (sorry, I couldn't find why) after setting headers, FormUrlEncodedContent was kinda omitted and the code simply didn't work.

Reply:
Ahhh, interesting.  I bet something is overwritten.  Sounds good.